internal/controller/dpuoperatorconfig_controller.go

```diff
@@ -184,6 +192,18 @@ func (r *DpuOperatorConfigReconciler) applyAllFromBinData(logger logr.Logger, bi
 	return nil
 }
 
+func (r *DpuOperatorConfigReconciler) ensureVSP(ctx context.Context, cfg *configv1.DpuOperatorConfig) error {
+	pi := platform.NewPlatformInfo()
```

Comment:
This code is currently called from the daemon, i.e. https://github.com/openshift/dpu-operator/blob/main/cmd/daemon/daemon.go#L57 which is running on the worker with the dpu installed.

This is being called here from the controller which is likely running on the control plane somewhere and not necessarily running on every dpu worker node. Are you sure the platform detection logic will work here?